drivers/tps6274x/tps6274x.c

```diff
@@ -54,15 +38,15 @@ unsigned int tps6274x_switch_voltage(unsigned int voltage)
     uint8_t vsel = (voltage - 1800) / 100;
     uint8_t vsel_not_set = 0;
     for(uint8_t i=0; i<4; i++){
-        /*get pin from struct, since all elements are of type gpio_t padding issues will be avoided by using sizeof*/
-        gpio_t pin = *((gpio_t*)((void*)&converter_config + i*sizeof(tps6274x_config_t)/5));
-        if(pin != GPIO_UNDEF) {
-            gpio_write(pin, (vsel & (0x01<<i)));
+        if(converter_config.vsel[i] != GPIO_UNDEF) {
+            gpio_write(converter_config.vsel[i], (vsel & (0x01<<i)));
             DEBUG("vsel%u :%u\n", i+1, (vsel & (0x01<<i)));
```

Comment:
This line will decide if the test runs. When DEBUG is enabled the test runs as expected. When DEBUG is disabled or this line is commented out RIOT will crash and restart. 
Does anybody know why this is happening?